Chinese Academy of Sciences CNIC CSTR.js

```diff
@@ -0,0 +1,194 @@
+{
+	"translatorID": "c070e5a2-4bfd-44bb-9b3c-4be20c50d0d9",
+	"label": "Chinese Academy of Sciences CNIC CSTR",
+	"creator": "cheney",
+	"target": "^https?://www\\.cstr\\.cn",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 97,
+	"inRepository": true,
+	"translatorType": 12,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2022-05-11 08:00:39"
+}
+
+function detectSearch(item) {
+	return !!item.CSTR;
+}
+
+function doSearch(item) {
+	
+	let url;
+	if (item.CSTR) {
+		url = "https://www.cstr.cn/openapi/v2/pid-cstr-service/detail?identifier=" + item.CSTR;
+	}
+	else if (item.query) {
+		url = "https://www.cstr.cn/openapi/v2/pid-common-service/statistics/metadata.search.aggregation?page=1&size=10&query=" + encodeURIComponent(item.query);
+	}
+	Z.debug(url);
+	ZU.doGet(url, parseJSON);
+}
+
+function detectWeb(doc, url) {
+	var searchRe = /^https?:\/\/(?:([^.]+\.))?(?:cstr\.cn)\/(?:search\/cstrDetail\/).*/;
+	if (searchRe.test(url)) {
+		return "journalArticle";	
+	}
+	else {
+		return "multiple";
+	}
+}
+// 解析页面元素获取元数据项目
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		var rows = doc.querySelectorAll('a.identifier');
+		if(rows){
+			var urls = [];
+			for (var i = 0; i < rows.length; i++) {
+				var id = rows[i].href.replace(/^https?:\/\/(?:([^.]+\.))?(?:cstr\.cn)\//,"");
+				urls.push('https://www.cstr.cn/openapi/v2/pid-cstr-service/detail?identifier=' + encodeURIComponent(id)
+				);
+			}
+			if(urls && urls.length > 0){
+				ZU.doGet(urls, parseJSON);
+			}
+		}		
+	}
+	else {
+		var id = doc.querySelector('a.identifier');		
+		if (id) {
+			id = id.innerText.replace(/^CSTR:\s*|v\d+|\s+.*$/ig, '');
+			var apiurl = 'https://www.cstr.cn/openapi/v2/pid-cstr-service/detail?identifier=' + encodeURIComponent(id);
+			ZU.doGet(apiurl, parseJSON);
+		}else{
+			if (!id) throw new Error('Could not find CSTR ID on page.');
+		}
+	}
+}
+
+function parseJSON(text) {
+	var obj = JSON.parse(text);
+	Z.debug(obj.code);
+
+	var code = obj.code;
+	if(code == 200){
+		var newItem = new Zotero.Item("journalArticle");
+		var content = obj.data.content;
+		newItem.title = content.resourceChineseName;
+		var identifier = content.identifier;
+		var creators = content.creators;
+		if (creators && creators.length) {
+			// arXiv.org format
+			for(var i =0 ; i<creators.length; i++){
+				// var creator_name = ZU.xpathText(creator, './/creator_name_cn').trim();
+				newItem.creators.push(
+					ZU.cleanAuthor(creators[i].creatorNameCN, "author", true)
+				);
+			}
+			
+		}
+		newItem.date = content.publicationDate;
+		var description = content.descriptionCN;
+		newItem.abstractNote = ZU.trimInternal(description);
+		newItem.notes.push({ note: identifier });
+		
+		newItem.extra = identifier;

```

Comment:
```suggestion
		newItem.extra = (newItem.extra || '') + '\nCSTR: ' + identifier;
```

`extra` should be key-value and existing content should be preserved.